src/grapple.c

```diff
@@ -11,6 +11,25 @@
 
 void SpawnBlood(vec3_t dest, float damage);
 
+float StartSpeed(vec3_t velocity)
+{
+	return sqrt(
+		velocity[0] * velocity[0] +
+		velocity[1] * velocity[1] +
+		velocity[2] * velocity[2]
+	);
+}
+
+float IncrementSpeed(float speed, float incr)
+{
+	return speed < (800 - incr) ? speed += incr : 800;
+}
+
+float DecrementSpeed(float speed, float decr)
+{
+	return speed > (800 + decr) ? speed -= decr : 800;
```

Comment:
same here. these one liners are super-geek, but it is hard to process them quickly...